src/Models/AdvectedPopulations/NPZD.jl

```diff
@@ -337,4 +340,9 @@ adapt_structure(to, npzd::NPZD) =
 
                                              adapt(to, npzd.particles))
 
+@inline nitrogen_flux(grid, advection, bgc::NPZD, tracers, i, j) = - (div_Uc(i, j, 0, grid, advection, biogeochemical_drift_velocity(bgc, Val(:D)), tracers.D) +
```

Comment:
I'm not convinced that this is generally true. I'm the column case it is since there is no horizontal velocity, but if the bottom cell didn't have a no slip boundary (or worse had some imposed flux which should give the cell below the boundary a value) then I think this will be wrong. Instead I think I need to borrow some code from a boundary condition to calculate the flux through the boundary properly.

Reply:
I'm a bit hesitant with indexing `k=0` here. This is a halo region... What's the thinking here?

Reply:
For tracers, the first halo cell is filled with the `k=1` value so `div_Uc`:
https://github.com/CliMA/Oceananigans.jl/blob/56fc69bcb08623424afe25e39ae616c29208095e/src/Advection/tracer_advection_operators.jl#L31C12-L33C103
becomes 
`1/Vᶜᶜᶜ(i, j, k, grid) * δzᵃᵃᶜ(i, j, k, grid, _advective_tracer_flux_z, advection, U.w, c) = ` -wC(k=0)Aᵃᵃᶜ/V. Whereas for the `k=1` it is w(C(k=1) - C(k=2))Aᵃᵃᶜ/V

The conservation tests don't pass with `k=1` either.

Reply:
I guess a more correct way to put this is:
https://github.com/CliMA/Oceananigans.jl/blob/56fc69bcb08623424afe25e39ae616c29208095e/src/Advection/centered_advective_fluxes.jl#L31C1-L31C1
?

Reply:
Written a better version of this now